packages/core/src/warnDuplicatePkg.ts

```diff
@@ -0,0 +1,28 @@
+interface Package {
+  from: string;
+  eager?: boolean;
+  loaded?: number;
+}
+
+interface Packages {
+  [key: string]: {
+    [key: string]: Package;
+  };
+}
+
+/**
+ * Warns applications using the shared scope if they have packages multiple times
+ */
+export const warnDuplicatePkg = (packages: Packages) => {
+  const entries = Object.entries(packages);
+
+  entries.forEach(([pkgName, versions]) => {
+    const instances = Object.keys(versions);
+    if (instances.length > 1) {
+      console.log(
```

Comment:
I preferred informing type it over warning type, since we tend to ignore warnings.

Reply:
Do you mean by default this needs to be enabled in development mode and then there should be an option to enable on production?

Reply:
>since we tend to ignore warnings.

That's unfortunately a developer choice, warnings have their place and we should respect the browser output. Some bundlers tend to or can be configured to remove `console.log` in production builds. So if someone opts in to show the errors in prod env, they might not be displayed. Warnings will stay. 

> Do you mean by default this needs to be enabled in development mode and then there should be an option to enable on production?

Correct